caffe2/CMakeLists.txt

```diff
@@ -149,50 +151,52 @@ if (FALSE)
 endif()
 
 # ---[ List of libraries to link with
-if(BUILD_CAFFE2)
+if (BUILD_CAFFE2)
   add_library(caffe2_protos STATIC $<TARGET_OBJECTS:Caffe_PROTO> $<TARGET_OBJECTS:Caffe2_PROTO>)
   add_dependencies(caffe2_protos Caffe_PROTO Caffe2_PROTO)
-  # If we are going to link protobuf locally inside caffe2 libraries, what we will do is
-  # to create a helper static library that always contains libprotobuf source files, and
-  # link the caffe2 related dependent libraries to it.
-  target_include_directories(caffe2_protos INTERFACE $<INSTALL_INTERFACE:include>)
-  # Reason for this public dependency is as follows:
-  # (1) Strictly speaking, we should not expose any Protobuf related functions. We should
-  #     only use function interfaces wrapped with our own public API, and link protobuf
-  #     locally.
-  # (2) However, currently across the Caffe2 codebase, we have extensive use of protobuf
-  #     functionalities. For example, not only libcaffe2.so uses it, but also other
-  #     binaries such as python extensions etc. As a result, we will have to have a
-  #     transitive dependency to libprotobuf.
-  #
-  # Good thing is that, if we specify CAFFE2_LINK_LOCAL_PROTOBUF, then we do not need to
-  # separately deploy protobuf binaries - libcaffe2.so will contain all functionalities
-  # one needs. One can verify this via ldd.
-  #
-  # TODO item in the future includes:
-  # (1) Enable using lite protobuf
-  # (2) Properly define public API that do not directly depend on protobuf itself.
-  # (3) Expose the libprotobuf.a file for dependent libraries to link to.
-  #
-  # What it means for users/developers?
-  # (1) Users: nothing affecting the users, other than the fact that CAFFE2_LINK_LOCAL_PROTOBUF
-  #     avoids the need to deploy protobuf.
-  # (2) Developers: if one simply uses core caffe2 functionality without using protobuf,
-  #     nothing changes. If one has a dependent library that uses protobuf, then one needs to
-  #     have the right protobuf version as well as linking to libprotobuf.a.
-  target_link_libraries(caffe2_protos PUBLIC protobuf::libprotobuf)
+else()
+  # Do not include caffe2 or caffe protos, but rather have it only be
+  # a library to attach local protobuf.
+  add_library(caffe2_protos STATIC utils/dummy.cpp)
```

Comment:
Out of curiosity - is this still needed, or is it a dangling temp test?

Reply:
(by the way, if you just need an interface library, you can create an INTERFACE target: https://cmake.org/cmake/help/v3.0/command/add_library.html)

Reply:
You read my mind. :) Unfortunately INTERFACE library didn't do it in this case - I tried.

Reply:
And just confirmed - result was

```
CMake Error at caffe2/CMakeLists.txt:190 (target_link_libraries):
  INTERFACE library can only be used with the INTERFACE keyword of
  target_link_libraries


CMake Error at cmake/public/utils.cmake:65 (message):
  You made a CMake build file error: target caffe2_protos must be of type
  either STATIC_LIBRARY or SHARED_LIBRARY.  However, I got INTERFACE_LIBRARY.
Call Stack (most recent call first):
  caffe2/CMakeLists.txt:194 (caffe2_interface_library)
```
There might be some way to eliminate the use of caffe2_interface_library, but I didn't see it being easy. Thoughts?